de.dentrassi.pm.maven/src/de/dentrassi/pm/maven/internal/MavenRepositoryChannelAggregator.java

```diff
@@ -174,76 +185,92 @@ private String makeUrl ( final String channelId )
         return result;
     }
 
-    private MavenInformation makeInfo ( final ArtifactInformation art, final Map<String, ArtifactInformation> map )
+    private Collection<MavenInformation> getInfos ( final ArtifactInformation art, final Map<String, ArtifactInformation> map )
     {
-        final MavenInformation info = new MavenInformation ();
+        final Collection<MavenInformation> infos = new LinkedList<> ();
+
+        /*
+         * Try to use the info of the artifact them-self.
+         */
 
         try
         {
+            final MavenInformation info = new MavenInformation ();
             MetaKeys.bind ( info, art.getMetaData () );
+            if ( info.getGroupId () != null & info.getArtifactId () != null && info.getVersion () != null )
+            {
+                // found direct meta data
+                infos.add ( info );
```

Comment:
At which time a direct meta data is found?
For POM files, the direct meta data should be available, so we could stop here as we need no childs for a pom.
Could someone add that meta data to a jar file, too?
For a jar file I would like to add all the child POMs as they could use different GAVs.